### ISSUE TITLE
crash

### ISSUE THREAD

**Issue by kin-fuyuki** [SquidDev-CC/CC-Tweaked#1609]
### Minecraft Version

1.19.x

### Version

1.108.0

### Details

the crash log:

```
java.lang.RuntimeException: Could not execute entrypoint stage 'client' due to errors, provided by 'moonlight'!
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:53)
	at net.minecraft.class_310.<init>(class_310.java:468)
	at net.minecraft.client.main.Main.main(Main.java:198)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:468)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71)
Caused by: java.lang.NoClassDefFoundError: dan200/computercraft/api/peripheral/IPeripheralProvider
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1012)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.defineClassFwd(KnotClassLoader.java:153)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:355)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at net.mehvahdjukaar.supplementaries.integration.CCCompat.makeSpeaker(CCCompat.java)
	at net.mehvahdjukaar.supplementaries.reg.ModRegistry.lambda$static$118(ModRegistry.java:581)
	at net.mehvahdjukaar.moonlight.api.platform.fabric.RegistryQueue$EntryWrapper.initialize(RegistryQueue.java:80)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at net.mehvahdjukaar.moonlight.api.platform.fabric.RegistryQueue.initializeEntries(RegistryQueue.java:43)
	at net.mehvahdjukaar.moonlight.api.platform.fabric.RegHelperImpl.lateRegisterEntries(RegHelperImpl.java:79)
	at net.mehvahdjukaar.moonlight.fabric.MoonlightFabric.commonSetup(MoonlightFabric.java:51)
	at net.mehvahdjukaar.moonlight.fabric.MoonlightFabricClient.onInitializeClient(MoonlightFabricClient.java:9)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:47)
	... 10 more
Caused by: java.lang.ClassNotFoundException: dan200.computercraft.api.peripheral.IPeripheralProvider
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:226)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
```

why it asks for for og computercraft libraries? how i fix it? i installed via prism launcher downloading from modrinth.

**Comment by fatboychummy** [comment]
Loading "og computercraft libraries" is completely fine, much of the code is still originally written by dan200, so that naming can't really be changed (or perhaps it can, but is left as is as a sort of tribute. I'm not sure about the legal stuff though, so I don't touch that).

However, the issue here seems to be that the mod Supplementaries is trying to load a class incorrectly. Looking through Supplementaries' 1.19 branch, I can see their last commit's (https://github.com/MehVahdJukaar/Supplementaries/commit/dcad07593cfb6ec655eb5398fb68caa84a1c98dd) changelog is "fixed cc compat", and they do remove the import for `IPeripheralProvider`. Thus, likely updating Supplementaries to the latest available for 1.19 (or removing it) will fix your issue.

**Comment by SquidDev** [comment]
Thank you for the report! As @fatboychummy says, this is an issue with Supplementaries trying to load Forge-specific classes on Fabric. It looks like that's been fixed, so updating to the latest version of supplementaries should fix it.

**Comment by kin-fuyuki** [comment]
i installed the newest supplementaries so might be a new bug, thx.

**Comment by SquidDev** [comment]
Ahhh, it looks like the version with the fix (2.4.16) made its way to Modrinth - it's only on CurseForge. You can [grab it from there instead](https://www.curseforge.com/minecraft/mc-mods/supplementaries/files/4620166)

**Comment by kin-fuyuki** [comment]
but this is for forge, im using fabric. i will talk to them tho.
